Reporter application

HwaEVE opened the pull request.

HwaEVE committed:
refactor(SecurityConfig): SecurityFilter 제거

HwaEVE committed:
feat(reporter): 기자 신청 기능 모듈 추가

HwaEVE committed:
refactor(member): 기존 MemberController/MemberService에서
기자 신청·변경 로직 제거

HwaEVE committed:
refactor(reporter): action 파라미터를 RequestStatus enum으로 변경

HwaEVE committed:
feat(reporter): 다중 신청 일괄 처리 기능 추가, 다중 상태 조회 쿼리 파라미터 지원

HwaEVE committed:
feat(reporter): 본인 신청 상태 조회 API 추가

HwaEVE committed:
feat(reporter): 반려가 되었으면 재신청을 할 수 있는 기능 추가

HwaEVE committed:
feat(test): 기자 신청·처리 API 통합 테스트 구현

HwaEVE committed:
feat(test): 테스트용 토큰에 대한 만료기한 설명 추가

HwaEVE committed:
feat(test): 관리자 대기·반려 상태 조회 API 테스트 추가

HwaEVE committed:
test: 관리자용 대기·반려 상태 조회 테스트 시나리오 추가

HwaEVE committed:
fix(test): 테스트 시나리오 신청, 중복 시 막기, 본인 조회 복구

HwaEVE committed:
fix(test): 일반 사용자2의 토큰의 정보를 저장이랑 일치 시킴

yulmucha commented on src/main/java/redlightBack/reporterapplication/domain/ReporterApplication.java:
```diff
@@ -0,0 +1,64 @@
+    @GeneratedValue(strategy = GenerationType.IDENTITY)
+    private Long id;
+
+    @OneToOne(fetch = FetchType.LAZY)
+    @JoinColumn(nullable = false)
+    private Member member;
```
거절되고 다시 신청하는 경우, 1)기존 데이터를 수정하는 것과 2)추가로 데이터를 만드는 것 중 어느 것이 더 나을까요?

yulmucha commented on src/main/java/redlightBack/reporterapplication/service/ReporterApplicationService.java:
```diff
@@ -0,0 +1,127 @@
+    private final ReporterApplicationRepository reporterApplicationRepository;
+    private final MemberRepository memberRepository;
+
+    public ReporterApplicationService(
+            ReporterApplicationRepository reporterApplicationRepository,
+            MemberRepository memberRepository
+    ) {
+        this.reporterApplicationRepository = reporterApplicationRepository;
+        this.memberRepository = memberRepository;
+    }
```
롬복의 `@RequiredArgsConstructor`를 활용하실 수 있습니다

yulmucha commented on src/main/java/redlightBack/reporterapplication/web/ReporterApplicationController.java:
```diff
@@ -0,0 +1,65 @@
+        return service.getMyApplication(userId);
+    }
+
+    // 3) 사용자 → 본인 재신청 (반려된 경우에만 가능)
+    @PutMapping("/me")
+    public ApplicationResponseDto resubmit(
```
신청과 재신청을 구분하지 않을 수도 있을 것 같네요

yulmucha commented on src/main/java/redlightBack/reporterapplication/web/ReporterApplicationController.java:
```diff
@@ -0,0 +1,65 @@
+    }
+
+    // 4) 관리자 → 다중 대기 & 반려 목록 조회
+    @GetMapping("/admin")
+    public List<ApplicationResponseDto> listByStatuses(
```
이러면 url이 `/reporter-applications/admin`이 되는데 "기자 지원서 중 관리자 지원서"와 같이 의미가 이상해지는 것 같네요.

yulmucha commented on src/main/java/redlightBack/reporterapplication/service/ReporterApplicationService.java:
```diff
@@ -0,0 +1,127 @@
+
+        if (reporterApplicationRepository.findByMember_UserId(userId).isPresent()) {
+            throw new ResponseStatusException(
+                    HttpStatus.CONFLICT, "이미 신청된 사용자입니다");
```
`HttpStatus.CONFLICT` ?

yulmucha replied:
그리고 이 방식이 `@RestcontrollerAdvice` 방식보다 나은지 고민해 보셨나요?
 (72번 줄이랑 같이 봐야 할 것 같네요)

yulmucha commented on src/main/java/redlightBack/reporterapplication/domain/ReporterApplication.java:
```diff
@@ -0,0 +1,64 @@
+
+    //반려 상태에서만 재신청
+    public void resubmit() {
+        if (this.status != RequestStatus.REJECTED) {
+            throw new IllegalStateException("REJECTED 상태에서만 재신청이 가능합니다");
+        }
```
비지니스 정책에 의해 바뀌는 부분이라고 볼 수도 있지 않을까요

yulmucha commented on src/main/java/redlightBack/reporterapplication/web/ReporterApplicationController.java:
```diff
@@ -0,0 +1,65 @@
+            @LoginMemberId String userId,
+            @RequestParam(defaultValue = "PENDING,REJECTED") List<RequestStatus> statuses
+    ) {
+        service.authorizeAdmin(userId);
```
관리자 권한 확인 메서드를 컨트롤러에서 호출해야 할까요?

HwaEVE committed:
refactor(reporter): 코드 리뷰 피드백 전반 반영

- 반려 기록 보존 위해 Member ↔ ReporterApplication 관계를 1:N 매핑으로 변경

- 롬복 @RequiredArgsConstructor로 생성자 자동 주입 적용

- 재신청용 PUT → POST 단일 엔드포인트로 통합

- 관리자 API URL을 /admin/reporter-applications 로 재배치

- HttpStatus.CONFLICT → BAD_REQUEST(400) 로 에러 코드 수정

- @RestControllerAdvice 기반 글로벌 예외 처리 적용

- 관리자 권한 검증 로직을 서비스 내부로 이동

HwaEVE committed:
refactor(admin): 관리자 콘솔 API 엔드포인트 경로를 /admin/console/reporter-applications 로 변경

HwaEVE committed:
refactor: SecurityConfig cors 설정

HwaEVE committed:
refactor: ReporterApplicationService import 최적화

HwaEVE committed:
refactor(common): GlobalExceptionHandler 에러 응답을 ResponseEntity<String>으로 변경

HwaEVE committed:
refactor(test): api 변경으로 인한 테스트 코드 url 변경

HwaEVE committed:
feat: 사용자의 관리자 여부에 관한 기능 추가

HwaEVE committed:
fix: WebConfig에 PATCH추가

HwaEVE committed:
fix: MemberController의 사용자 권한을 확인하는 기능을 수정함.

HwaEVE committed:
fix: Member의 로직 삭제

HwaEVE committed:
refactor: ReporterApplicationService의 다중승인&반려에서 반려가 된건 승인이 불가하도록 로직 변경

HwaEVE committed:
refactor: 관리자가 기자를 조회할 시 닉네임 까지 조회가 가능하게 변경

HwaEVE committed:
feat: 반려 처리된 기자 신청서를 재승인 불가 하는 기능 테스트 코드 추가

HwaEVE committed:
Merge branch 'main' into reporterApplication
